keanu-project/src/main/java/io/improbable/keanu/vertices/bool/nonprobabilistic/operators/BoolModelResultVertex.java

```diff
@@ -0,0 +1,42 @@
+package io.improbable.keanu.vertices.bool.nonprobabilistic.operators;
+
+import io.improbable.keanu.tensor.bool.BooleanTensor;
+import io.improbable.keanu.vertices.NonProbabilistic;
+import io.improbable.keanu.vertices.Vertex;
+import io.improbable.keanu.vertices.VertexLabel;
+import io.improbable.keanu.vertices.bool.BoolVertex;
+import io.improbable.keanu.vertices.dbl.KeanuRandom;
+import io.improbable.keanu.vertices.dbl.nonprobabilistic.operators.multiple.ModelVertex;
+
+/**
+ * A non-probabilistic boolean vertex whose value is extracted from an upstream model vertex.
+ */
+public class BoolModelResultVertex extends BoolVertex implements NonProbabilistic<BooleanTensor> {
```

Comment:
Having thought more on this, I wonder if the Result Vertices would be better as abstract base classes that each different model Vertex can provides a concrete implementation of (albeit a version that can read directly from a map would likely be useful for a lot of cases).  For instance, I could imagine a model that builds up a huge database - we wouldn't necessary want to store everything in memory all the time and we might just want to retrieve from disk as and when required?  Similarly, for something like a loop Vertex we'd be able to read the value directly from the "output" of the Loop whereas this implementation would mandate we store this output in a map to allow it to be extracted.

Reply:
Interesting idea - it sounds like a separate ticket and outside the scope of this ticket though. I think it would be best to get a basic implementation completed first that used maps and then add a ticket to extend this to use a user defined concept